apps/dashboard/src/components/workflow-editor/steps/in-app/in-app-editor.tsx

```diff
@@ -0,0 +1,50 @@
+import { RiPencilRuler2Line } from 'react-icons/ri';
+import { UiSchemaGroupEnum, type UiSchema } from '@novu/shared';
+
+import { getComponentByType } from '@/components/workflow-editor/steps/component-utils';
+
+const avatarKey = 'avatar';
+const subjectKey = 'subject';
+const bodyKey = 'body';
+const redirectKey = 'redirect';
+const primaryActionKey = 'primaryAction';
+const secondaryActionKey = 'secondaryAction';
+
+export const InAppEditor = ({ uiSchema }: { uiSchema?: UiSchema }) => {
+  if (!uiSchema || uiSchema?.group !== UiSchemaGroupEnum.IN_APP) {
+    return null;
+  }
+
+  const {
+    [avatarKey]: avatar,
+    [subjectKey]: subject,
+    [bodyKey]: body,
+    [redirectKey]: redirect,
+    [primaryActionKey]: primaryAction,
+    [secondaryActionKey]: secondaryAction,
+  } = uiSchema.properties ?? {};
+
+  return (
+    <div className="flex flex-col gap-3">
+      <div className="flex items-center gap-2.5">
+        <RiPencilRuler2Line className="text-foreground-950 size-5 p-0.5 text-sm font-medium" />
+        <span>In-app Template</span>
+      </div>
+      <div className="flex flex-col gap-1 rounded-xl border border-neutral-100 p-1">
+        <div className="flex gap-1">
+          {avatar && getComponentByType({ component: avatar.component })}
+          {subject && getComponentByType({ component: subject.component })}
+        </div>
+        {body && getComponentByType({ component: body.component })}
```

Comment:
<!-- Notes are always non-blocking and simply highlight something the reader should take note of.  -->
🗒 **note (non-blocking):** We might need to handle the case where the `Subject` will be missing‏

Reply:
Correct. If both avatar and subject are missing their wrapping div should also be removed as well.

Reply:
> We might need to handle the case where the Subject will be missing and move the body next to avatar in that case

@BiswaViraj I don't think that moving the body up in that case is correct, because it will create a "gap" on the left side of the form. I would rather keep the `avatar` and `body` in separate rows.

However, we can support what @SokratisVidros is saying and remove unnecessary div.

Reply:
![Screenshot 2024-11-07 at 11 27 45](https://github.com/user-attachments/assets/8d459cb7-88ec-45f8-acae-00341468995b)

@BiswaViraj check the screenshot, also to cover that case it will bloat the code a bit

Reply:
Ok got it, my bad confused it with the Preview 🤦🏼 
For the editor all the fields will be available always.